client/webserver/webserver.go

```diff
@@ -231,7 +230,7 @@ func (s *WebServer) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	// Update the listening address in case a :0 was provided.
 	s.addr = listener.Addr().String()
 
-	s.wsServer.SetContext(ctx)
+	s.wsServer.Run(ctx)
```

Comment:
So we're a little off here, but let's go with it for now.  I'm planning to look into using the [`http.Server.BaseContext`](https://golang.org/pkg/net/http/#Server) field to follow a slightly more idiomatic approach to request scoped contexts.  Basically `(*Server).HandleConnect` pulls the context out via `r.Context()` and sends that into `(*Server).connect` as an input arg to be ultimately used to manage the websocket client messaging loops.  It passes the context around more, kinda needlessly, but it's still something I'm considering and also not in scope for this PR.